SharpPcap/LibPcap/LibPcapLiveDevice.cs

```diff
@@ -156,67 +180,89 @@ public override void Open(DeviceConfiguration configuration)
+                    }
 
-                if (configuration.Immediate.HasValue)
+                    var activationResult = LibPcapSafeNativeMethods.pcap_activate(Handle);
+                    if (activationResult < 0)
+                    {
+                        string err = "Unable to activate the adapter (" + Name + "). Return code: " + activationResult.ToString();
+                        throw new PcapException(err);
+                    }
+                }
+                else
                 {
-                    if (!immediate_supported && !mintocopy_supported)
+                    // Shall we silently ignore unsupported configuration parameters
+                    // TimestampResolution, TimestampType, Monitor and BufferSize
+                    // or check and throw exception?
+
+                    //open will activate
+                    if ( null != credentials)
                     {
-                        configuration.RaiseConfigurationFailed(
-                            nameof(configuration.Immediate), 
-                            (int)PcapError.PlatformNotSupported
-                        );
+                        // We got authentication, so this is an rpcap device
+                        var auth = RemoteAuthentication.CreateAuth(credentials);
+                        Handle = LibPcapSafeNativeMethods.pcap_open(
+                            Name,                               // name of the device
+                            configuration.Snaplen,              // portion of the packet to capture.
+                            (short)mode,                        // flags
+                            (short)configuration.ReadTimeout,   // read timeout
+                            ref auth,                           // authentication
+                            errbuf);                            // error buffer
                     }
-                    else if (immediate_supported)
+                    else
                     {
-                        var immediate = configuration.Immediate.Value ? 1 : 0;
-                        Configure(
-                            configuration, nameof(configuration.Immediate),
-                            LibPcapSafeNativeMethods.pcap_set_immediate_mode, immediate
-                        );
+                        //We got no authentication, but modes other than 
+                        // DeviceModes.Promiscuous and DeviceModes.MaxResponsiveness require pcap_open()
+                        Handle = LibPcapSafeNativeMethods.pcap_open(
+                            Name,                               // name of the device
+                            configuration.Snaplen,              // portion of the packet to capture.
+                            (short)mode,                        // flags
+                            (short)configuration.ReadTimeout,   // read timeout
+                            IntPtr.Zero,                        // authentication
```

Comment:
You don't need this version of `pcap_open` where you pass `IntPtr.Zero`

Reason being `RemoteAuthentication.CreateAuth` accept null credentials as input without issues.